src/data_repr.rs

```diff
@@ -55,6 +57,35 @@ impl<A> OwnedRepr<A> {
         self.ptr
     }
 
+    /// Return end pointer
+    pub(crate) fn as_end_nonnull(&self) -> NonNull<A> {
```

Comment:
I'll leave this for now, I forgot about it, that's mostly why. And that the as_ convention is weirdly preferring "as_" for pointer accessors, so that's what a couple of our methods have. I don't even like `.as_ptr()` - it should normally be just `.ptr()` etc.